plugins/package-managers/nuget/src/main/kotlin/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,304 @@
+        val errors: List<String>,
+        val warnings: List<String>
+    )
+
+    @Serializable
+    internal class Party(
+        val type: String,
+        val role: String,
+        val name: String?,
+        val email: String?,
+        val url: String?
+    )
+}
+
+private const val TYPE = "NuGet"
+
+private fun List<NuGetInspector.Party>.toAuthors(): SortedSet<String> =
+    filter { it.role == "author" }.mapNotNullTo(sortedSetOf()) { party ->
+        buildString {
+            party.name?.let { append(it) }
+            party.email?.let {
+                append(if (party.name != null && it.isNotBlank()) " <$it>" else it)
+            }
+        }.takeIf { it.isNotBlank() }
+    }
+
+private fun collectErrorMessage(result: NuGetInspector.Result): String {
+    var errorMessage = ""
+    result.headers.first().errors.forEach {
+        errorMessage += it + "\n"
+    }
+    result.packages.first().errors.forEach {
+        errorMessage += it + "\n"
+    }
+    result.dependencies.forEach { dep -> dep.errors.forEach { errorMessage += it + "\n" } }
+    return errorMessage
+}
+
+internal fun NuGetInspector.Result.toOrtProject(
+    managerName: String,
+    analysisRoot: File,
+    definitionFile: File,
+): Project {
+    val id = Identifier(
+        type = managerName,
+        namespace = "",
+        name = definitionFile.relativeTo(analysisRoot).invariantSeparatorsPath,
+        version = ""
+    )
+    val workingDir = definitionFile.parentFile
+
+    val nestedPackages: MutableList<NuGetInspector.PackageData> = mutableListOf()
+
+    packages.forEach {
+        it.dependencies.forEach { pkg -> nestedPackages.add(pkg) }
+    }
+
+    val packageReferences = nestedPackages.toPackageReferences()
+
+    val scopes = sortedSetOf(Scope(headers.first().projectFramework, packageReferences))
+    return Project(
+        id = id,
+        definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+        vcs = VcsInfo.EMPTY,
+        authors = setOf(),
+        vcsProcessed = PackageManager.processProjectVcs(workingDir),
+        declaredLicenses = setOf(),
+        homepageUrl = "",
+        scopeDependencies = scopes
+    )
+}
+
+private fun List<NuGetInspector.PackageData>.toPackageReferences(): SortedSet<PackageReference> =
+    mapTo(sortedSetOf()) { it.toPackageReference() }
+
+private fun NuGetInspector.PackageData.toPackageReference() =
+    PackageReference(
+        id = Identifier(type = TYPE, namespace = "", name = name, version = version ?: ""),
+        dependencies = dependencies.toPackageReferences()
+    )
+
+internal fun Set<NuGetInspector.PackageData>.toOrtPackages(): Set<Package> =
+    groupBy { "${it.name}:${it.version}" }.mapTo(mutableSetOf()) { (_, packages) ->
+        val pkg = packages.first()
+
+        fun NuGetInspector.PackageData.getHash(): Hash = Hash.create(
+            sha512?.lowercase() ?: sha256?.lowercase() ?: sha1?.lowercase() ?: md5?.lowercase() ?: ""
```

Comment:
Technically, lower-casing should not be needed. Is it just for beautification / comparison? Anyway, if at all, do it only once in the end:

    (sha512 ?: sha256 ?: sha1 ?: md5 ?: "").lowercase()